pkg/cloud/cloud.go

```diff
@@ -1182,15 +1169,23 @@ func (c *cloud) checkDesiredSize(ctx context.Context, volumeID string, newSizeGi
 	}
 
 	// AWS resizes in chunks of GiB (not GB)
-	oldSizeGiB := aws.Int64Value(volume.Size)
-	if oldSizeGiB >= newSizeGiB {
-		return oldSizeGiB, nil
+	realSizeGiB := aws.Int64Value(volume.Size)
+
+	if realSizeGiB < desiredSizeGiB {
```

Comment:
Nit: This chunk was a bit hard to read on first glance, could you add a comment or refactor this into a helper function? 

Something Like "modifyVolumeIdempotencyCheck"

Reply:
Isn't that already what this function is? I don't think making a function that will never be reused is a good idea, I'll go with your second suggestion and add a comment unless there are other objections.